zipline/utils/pandas_utils.py

```diff
@@ -320,3 +320,30 @@ def empty_dataframe(*columns):
     dtype: object
     """
     return pd.DataFrame(np.array([], dtype=list(columns)))
+
+
+def check_indexes_all_same(indexes, message="Indexes are not equal."):
+    """Check that a list of Index objects are all equal.
+
+    Parameters
+    ----------
+    indexes : iterable[pd.Index]
+        Iterable of indexes to check.
+
+    Raises
+    ------
+    ValueError
+        If the indexes are not all the same.
+    """
+    iterator = iter(indexes)
+    first = next(iterator)
```

Comment:
should we support vacuous truth here?

Reply:
I can't think of a case where I'd want to call this with no inputs (or, for that matter, with one input, though that does currently work). I think this is different from `all` and `any` (which do support an empty input) because this is checking that a binary relation holds between elements in the input, vs. all/any which are checking a simple predicate on each input.